update: update the Kotlin/Native Gradle tutorial (#2761)
JetBrains/kotlin-web-site: docs/topics/native/native-gradle.md

[Gradle](https://gradle.org) is a build system that is very commonly used in the Java, Android, and other ecosystems. It is the default choice for Kotlin/Native and Multiplatform
when it comes to build systems.

> While most IDEs including [IntelliJ IDEA](https://www.jetbrains.com/idea) can generate necessary Gradle files,

Reviewer: ```suggestion
While most IDEs, including [IntelliJ IDEA](https://www.jetbrains.com/idea), can generate necessary Gradle files,
```

> this tutorial covers how to create necessary files manually, to provide a better understanding of how things work under the hood.

Reviewer: ```suggestion
this tutorial covers how to create them manually to provide a better understanding of how things work under the hood.
```

<tabs group="build-script">
<tab title="Kotlin" group-key="kotlin">

   ```kotlin
   // build.gradle.kts
   plugins {
       kotlin("multiplatform") version "%kotlinVersion%"
   }

   repositories {
       mavenCentral()
   }

   kotlin {
     macosX64("native") { // on macOS
     // linuxX64("native") // on Linux
     // mingwX64("native") // on Windows
       binaries {
         executable()
       }
     }
   }

   tasks.withType<Wrapper> {
     gradleVersion = "%gradleVersion%"
     distributionType = Wrapper.DistributionType.BIN
   }
   ```

</tab>
<tab title="Groovy" group-key="groovy">

   ```groovy
   // build.gradle
   plugins {
       id 'org.jetbrains.kotlin.multiplatform' version '%kotlinVersion%'
   }

   repositories {
       mavenCentral()
   }

   kotlin {
     macosX64('native') { // on macOS
     // linuxX64('native') // on Linux
     // mingwX64('native') // on Windows
       binaries {
         executable()
       }
     }
   }

   wrapper {
     gradleVersion = '%gradleVersion%'
     distributionType = 'BIN'
   }
   ```

</tab>
</tabs>

   You can use different [target presets](mpp-supported-platforms.md), such as `macosX64`, `mingwX64`, `linuxX64`, `iosX64`,

>    to define corresponding the target platform. The preset name describes a platform for which you are compiling your code.

Reviewer: ```suggestion
   to define the corresponding target platform. The preset name describes a platform for which you are compiling your code.
```

   These target presets optionally take the target name as a parameter, which is `native` in this case.
   The target name is used to generate the source paths and task names in the project.

3. Create an empty `settings.gradle` or `settings.gradle.kts` file in the project folder.

4. Create a folder `src/nativeMain/kotlin` and place inside the `hello.kt` file with the following content:
   
   ```kotlin
   fun main() {
     println("Hello, Kotlin/Native!")
   }
   ```

   By convention, all sources are located in the `src/<target name>[Main|Test]/kotlin` folders, where `main` is for the source code

>    and `test` is for tests. `<target name>` corresponds to the target platform (in this case `native`), as specified in the build file.

Reviewer 2: Is it correct to call the given optional name ('native') a target platform?

Reviewer 2: Maybe expand the explanation and add an example if how it looks when the name is not specified

Author: Decided to come back to it later

> Now you are ready build your project and run the application. 

Reviewer: ```suggestion
Now you are ready to build your project and run the application. 
```

## Build and run the application

1. From the root project folder, run the build command:

   ```bash
   gradle nativeBinaries
   ```

>    This command creates a folder `build/bin/native` and two subfolders inside: `debugExecutable` and `releaseExecutable`. They contain corresponding binary files.  

Reviewer 2: "Folder" is a bit Windows-specific and belongs to the GUI concept; in the console-based tutorial the more general term `directory` would fit better.

   By default, the name of the binary file is the same as the project folder. 

3. To run the project, execute the following command:

   ```bash

>    build/bin/native/debugExecutable/<your_project_folder_name>.kexe

Reviewer 2: `your_project_folder_name` seems a bit long and confusing (like: "why the _folder_ name"?). Maybe just `project_name` if it's correct.

   ```

   Terminal prints "Hello, Kotlin/Native!".

## Open the project in an IDE

> Now you can open your project in any IDE that supports Gradle. If you use IntelliJ IDEA:

Reviewer 2: Maybe some "selling" intro? Something about powerful tools and the convenience of the IDE. Just an idea, no pressure)

Reviewer 2: And a link to K/N + IDEA tutorial. Maybe in the What's next section.

Author: There was a link to the K/N+IDEA tutorial in the intro. 
Don't think that mention it twice is a good idea.

But I like your "selling" intro point :)
